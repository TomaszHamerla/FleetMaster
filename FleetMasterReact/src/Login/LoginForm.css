/* LoginForm.css */

.login-form-container {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
}

.login-form {
  width: 400px;
  padding: 30px;
  border: 1px solid #ddd;
  border-radius: 5px;
  background-color: #fff;
  box-shadow: 0px 0px 10px 0px rgba(0, 0, 0, 0.1);
}

.login-form h2 {
  margin-bottom: 20px;
  text-align: center;
}

.form-group {
  margin-bottom: 15px;
}

.form-label {
  display: block;
  margin-bottom: 5px;
}

.form-input {
  width: 100%;
  padding: 10px;
  border: 1px solid #ddd;
  border-radius: 5px;
}
.buttons{
display: flex;
justify-content: center;
}
.form-button {
  padding: 10px 40px;
  margin: 5px;
  border: none;
  border-radius: 5px;
  background-color: #007bff;
  color: #fff;
  cursor: pointer;
}
.form-button-gray{
  padding: 10px 40px;
  margin: 5px;
  border: none;
  border-radius: 5px;
  background-color: #404346;
  color: #fff;
  cursor: pointer;
}

.form-button:hover {
  background-color: #0056b3;
}
@media (max-width: 600px) {
  .login-form {
    width: 100%;
    padding: 15px;
  }
}
