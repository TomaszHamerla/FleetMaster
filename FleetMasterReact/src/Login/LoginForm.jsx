import React, { useState } from "react";
import "./LoginForm.css";

const LoginForm = () => {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");

  const handleUsernameChange = (event) => {
    setUsername(event.target.value);
  };

  const handlePasswordChange = (event) => {
    setPassword(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    console.log("Submitted:", { username, password });
  };

  return (
    <div className="login-form-container">
      <form onSubmit={handleSubmit} className="login-form">
        <h2>Login</h2>
        <div className="form-group">
          <label htmlFor="username" className="form-label">
            Username:
          </label>
          <input
            placeholder="Username"
            type="text"
            id="username"
            value={username}
            onChange={handleUsernameChange}
            className="form-input"
            required
          />
        </div>
        <div className="form-group">
          <label htmlFor="password" className="form-label">
            Password:
          </label>
          <input
            placeholder="Password"
            type="password"
            id="password"
            value={password}
            onChange={handlePasswordChange}
            className="form-input"
            required
          />
        </div>
        <button type="submit" className="form-button">
          Log In
        </button>
      </form>
    </div>
  );
};

export default LoginForm;
