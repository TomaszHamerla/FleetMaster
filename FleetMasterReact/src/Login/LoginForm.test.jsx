import '../../node_modules/@testing-library/jest-dom/'

import React from "react";
import { render } from '@testing-library/react';
import LoginForm from './LoginForm';

test('LoginForm should match snapshot',()=>{
    const { asFragment } = render(<LoginForm/>);
    expect(asFragment()).toMatchSnapshot();
});
