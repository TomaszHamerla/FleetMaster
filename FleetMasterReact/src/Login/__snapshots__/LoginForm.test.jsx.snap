// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="login-form-container"
  >
    <form
      class="login-form"
    >
      <h2>
        Login
      </h2>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="username"
        >
          Username:
        </label>
        <input
          class="form-input"
          id="username"
          placeholder="Username"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="password"
        >
          Password:
        </label>
        <input
          class="form-input"
          id="password"
          placeholder="Password"
          required=""
          type="password"
          value=""
        />
      </div>
      <div
        class="buttons"
      >
        <button
          class="form-button-gray"
          type="submit"
        >
          Login
        </button>
        <button
          class="form-button"
          type="submit"
        >
          Sign up
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
